src/OpenLoco/src/Objects/ObjectManager.cpp

```diff
@@ -983,8 +987,72 @@ namespace OpenLoco::ObjectManager
     // 0x004C3A9E
     void updateYearly2()
     {
-        // update available vehicles/roads/airports/etc.
-        call(0x004C3A9E);
+        const auto currentYear = getCurrentYear(); // 0x004C3AB7
+
+        // 0x004C3A9E, 004C3B8E-004C3B95
+        for (uint16_t vehicleObjId = 0; vehicleObjId < ObjectManager::getMaxObjects(ObjectType::vehicle); vehicleObjId++)
+        {
+            const auto* vehicleObject = ObjectManager::get<VehicleObject>(vehicleObjId); // 004C3AA0
+
+            if (vehicleObject == nullptr) // 004C3AA7
+            {
+                continue; // 0x004C3AAA
+            }
+
+            if (currentYear == vehicleObject->designed) // 004C3AB0-004C3ABE
+            {
+                for (Company& company : CompanyManager::companies()) // 004C3AC4-004C3AC9, 004C3B44-004C3B4F
+                {
+                    if (company.name == 0) // 004C3ACB-004C3ACF
+                    {
+                        continue;
+                    }
+
+                    // 004C3AD1-004C3AE7
+                    const auto forbiddenVehicles = CompanyManager::isPlayerCompany(company.id()) ? getGameState().forbiddenVehiclesPlayers : getGameState().forbiddenVehiclesCompetitors;
+                    if (forbiddenVehicles & (1U << enumValue(vehicleObject->type))) // 004C3AED-004C3AF6
+                    {
+                        continue;
+                    }
+
+                    if (company.unlockedVehicles[vehicleObjId]) // 004C3AF8-004C3B07
+                    {
+                        continue;
+                    }
+
+                    company.unlockedVehicles.set(vehicleObjId, true);                                    // 004C3B02
+                    company.availableVehicles = VehicleManager::determineAvailableVehicleTypes(company); // 004C3B09-004C3B13
+
+                    if (!CompanyManager::isPlayerCompany(company.id())) // 004C3B14-004C3B1A
```

Comment:
whilst L1012 was an isPlayerCompany check this one is actually a `CompanyManager::getControllingId() != company.id()` its subtly different and it means that you don't post a message when your networked competitor unlocks a vehicle.